add section on Git submodules (#2084)
vercel/docs: pages/docs/git-integrations/index.mdx

Additionally, monorepos are currently only supported with the [GitHub Integration](/github). Support for [GitLab](/gitlab) and [Bitbucket](/bitbucket) are upcoming.

## Git Submodules

> Deploying Git submodules are supported as long as they are publically accessible and are accessed via the HTTP protocol.

Reviewer: ```suggestion
Deploying Git submodules with the [Git Integration](/docs/git-integrations) is supported as long as the submodule is publicly accessible via the HTTP protocol.
```

Author: Correct.

> Git submodules that are private or requested over SSH will fail to get retrieved.

Reviewer 2: Pretty confident that's the case, previously they'd fail silently.
